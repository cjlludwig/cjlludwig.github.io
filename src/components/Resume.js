import { Document, Page, pdfjs } from 'react-pdf'
import { useState } from 'react';

import '../static/components/Resume.css'
import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
import 'react-pdf/dist/esm/Page/TextLayer.css';

import resumePdf from '../static/images/cjludwig_resume.pdf'

pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

function Resume() {
  const [numPages, setNumPages] = useState(null);
  const [pageNumber, setPageNumber] = useState(1);

  function onDocumentLoadSuccess({ numPages }) {
    setNumPages(numPages);
    setPageNumber(1);
  }

  function changePage(offset) {
    setPageNumber(prevPageNumber => prevPageNumber + offset);
  }

  function previousPage() {
    changePage(-1);
  }

  function nextPage() {
    changePage(1);
  }

  return (
    <>
      <Document file={resumePdf} onLoadSuccess={onDocumentLoadSuccess}>
        <Page 
          key={1}
          className={pageNumber === 1 ? "prevPage" : ""}
          pageNumber={1} 
        />
        <Page
          key={2}
          pageNumber={2}
          className={pageNumber === 2 ? "prevPage" : ""}
        />
      </Document>
      <div className='pdf-buttons'>
        <button
          type="button"
          disabled={pageNumber <= 1}
          onClick={previousPage}
          >
          {"<"}
        </button>
        <span>
          {pageNumber || (numPages ? 1 : '--')} of {numPages || '--'}
        </span>
        <button
          type="button"
          disabled={pageNumber >= numPages}
          onClick={nextPage}
        >
          {">"}
        </button>
      </div>
    </>
  );
}

export default Resume;