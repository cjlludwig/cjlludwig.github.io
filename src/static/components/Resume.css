body {
  margin: 0;
  background-color: #525659;
  font-family: Segoe UI, Tahoma, sans-serif;
}

.Example input,
.Example button {
  font: inherit;
}

.Example header {
  background-color: #323639;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.5);
  padding: 20px;
  color: white;
}

.Example header h1 {
  font-size: inherit;
  margin: 0;
}

.Example__container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 10px 0;
  padding: 10px;
}

.Example__container__load {
  margin-top: 1em;
  color: white;
}

.react-pdf__Document {
  margin: 1em 0;

  display: flex;
  flex-direction: column;
  align-items: center;
}

.react-pdf__Page {
  max-width: calc(100% - 2em);
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.5);
  margin: 1em;
}

.react-pdf__Page canvas {
  max-width: 100%;
  height: auto !important;
  border-radius: 8px;
  box-shadow: 0 30px 40px 0 rgb(16 36 94 / 20%);
}

.react-pdf__message {
  padding: 20px;
  color: white;
}

.react-pdf__Page__canvas {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.pdf-buttons {
  text-align: center;
  box-shadow: 0 10px 60px 0 rgb(16 36 94 / 20%);
  border-radius: 4px;
  background: white;
  display: table;
  margin-left: auto;
  margin-right: auto;

  position: absolute;
  left: 47%;
  bottom: 5%;
}

.pdf-buttons button {
  width: 44px;
  height: 44px;
  background: white;
  border: 0;
  font: inherit;
  font-size: 0.8em;
  border-radius: 4px;
}

.page-controls span {
  font: inherit;
  font-size: 0.8em;
  padding: 0 0.5em;
}
