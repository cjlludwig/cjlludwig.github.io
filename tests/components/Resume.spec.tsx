import React from "react";
import renderer from "react-test-renderer";

import Resume from "../../src/components/Resume";

describe("<Resume />", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<Resume />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});