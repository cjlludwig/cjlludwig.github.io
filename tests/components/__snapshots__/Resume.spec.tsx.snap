// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Resume /> renders correctly 1`] = `
[
  <div
    className="react-pdf__Document"
  >
    <div
      className="react-pdf__message react-pdf__message--no-data"
    >
      No PDF file specified.
    </div>
  </div>,
  <div
    className="pdf-buttons"
  >
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &lt;
    </button>
    <span>
      1
       of 
      --
    </span>
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &gt;
    </button>
  </div>,
]
`;
